app/lib/common/providers/notifiers/client_pref_notifier.dart

```diff
@@ -0,0 +1,218 @@
+/// Based on https://github.com/gamako/shared_preferences_riverpod/blob/master/lib/shared_preferences_riverpod.dart
+/// but bound to the specific client deviceIDs;
+library;
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk.dart';
+import 'package:riverpod/riverpod.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+/// The type parameter `T` is the type of value that will
+/// be persisted in [SharedPreferences].
+///
+/// To update the value, use the [update()] function.
+/// Direct assignment to state cannot be used.
+///
+/// ```dart
+/// await watch(booPrefProvider.notifier).update(v);
+/// ```
+///
+///
+class PrefNotifier<T> extends StateNotifier<T> {
+  PrefNotifier(this.prefKey, this.defaultValue) : super(defaultValue) {
+    _init();
+  }
+
+  late SharedPreferences prefs;
+  String prefKey;
+  T defaultValue;
+
+  void _init() async {
+    prefs = await sharedPrefs();
+    final newValue = (prefs.get(prefKey) as T? ?? defaultValue);
+    state = newValue;
+  }
+
+  /// Updates the value asynchronously.
+  Future<void> update(T value) async {
+    if (value is String) {
+      await prefs.setString(prefKey, value);
+    } else if (value is bool) {
+      await prefs.setBool(prefKey, value);
+    } else if (value is int) {
+      await prefs.setInt(prefKey, value);
+    } else if (value is double) {
+      await prefs.setDouble(prefKey, value);
+    } else if (value is List<String>) {
+      await prefs.setStringList(prefKey, value);
+    }
+    super.state = value;
+  }
+
+  /// Do not use the setter for state.
+  /// Instead, use `await update(value).`
+  @override
+  set state(T value) {
+    assert(false,
+        "Don't use the setter for state. Instead use `await update(value)`.",);
+    Future(() async {
+      await update(value);
+    });
+  }
+}
+
+/// Returns the [Provider] that has access to the value of preferences.
```

Comment:
A lot of commented code here. We can remove this!

Reply:
These are doc-comments. The code is example code for the generated docs. It comes from the copy-paste I made.